Geometry/VeryForwardGeometryBuilder/src/DetGeomDesc.cc

```diff
@@ -27,7 +28,14 @@ DetGeomDesc::DetGeomDesc(DDFilteredView* fv)
       m_name(((fv->logicalPart()).ddname()).name()),
       m_params(((fv->logicalPart()).solid()).parameters()),
       m_copy(fv->copyno()),
-      m_z(fv->geoHistory().back().absTranslation().z()) {}
+      m_z(fv->geoHistory().back().absTranslation().z()),
+      m_sensorType("") {
+  std::string sensor_name = fv->geoHistory().back().logicalPart().name().fullname();
+  std::size_t found = sensor_name.find(DDD_CTPPS_PIXELS_SENSOR_NAME);
+  if (found != std::string::npos && sensor_name.substr(found - 4, 3) == DDD_CTPPS_PIXELS_SENSOR_TYPE_2x2) {
+    m_sensorType = DDD_CTPPS_PIXELS_SENSOR_TYPE_2x2;
+  }
+}
```

Comment:
It would be good to create a second version of this constructor that takes the new DD4hep `cms::DDFilteredView`, either in this PR or a follow-up PR. We are at the end of the DD4hep migration, so this code that uses old DD will need to be eliminated in a few months.

Reply:
Absolutely correct! The DD4Hep migration (and reco-geometry storage in DB) have been started in parallel. @malbouis is leading this task. The ETA is March 2020 - is this OK? Our roadmap is sketched here: https://indico.cern.ch/event/866706/contributions/3652254/attachments/1952609/3242097/pps_geometry_dd4hep.pdf .

Reply:
@cvuosalo , I think, that migration to DD4hep should be done in following PRs not here.

Reply:
@jan-kaspar Thanks for the info. That is very helpful.